src/target/riscv/riscv-011.c

```diff
@@ -2371,7 +2385,9 @@ static int init_target(struct command_context *cmd_ctx,
 {
 	LOG_DEBUG("init");
 	RISCV_INFO(generic_info);
+	/* TODO: replace read and write with single access function*/
 	generic_info->read_memory = read_memory;
+	generic_info->write_memory = write_memory;
```

Comment:
Thanks! I got you. One of my next MRs replaces high-level read/write functions to a combined one. It will decrease amount of copy-paste. I think it also will be able to be called directly